communication/src/TlsConnectionManager.cpp

```diff
@@ -311,24 +313,35 @@ void ConnectionManager::resolve(NodeNum i) {
 }
 
 void ConnectionManager::connect(NodeNum i, const asio::ip::tcp::endpoint& endpoint) {
-  auto [it, inserted] = connecting_.emplace(i, asio::ip::tcp::socket(io_context_));
+  auto [it, inserted] =
+      connecting_.emplace(i, std::make_pair(asio::ip::tcp::socket(io_context_), asio::steady_timer(io_context_)));
   ConcordAssert(inserted);
   status_->num_connecting = connecting_.size();
-  it->second.async_connect(
+  LOG_DEBUG(logger_, "connecting to node : " << i);
+  // If async_connect takes too long to finish
+  // the timeout will happen after given duration
+  connecting_.at(i).second.expires_after(CONNECT_DEADLINE);
```

Comment:
Shouldnt the async_wait call for the timer be here

Reply:
its there at line 343 in code

Reply:
it is there at line 343 in code

Reply:
Yes. But shouldnt that be moved to here logically

Reply:
it works in the fashion where at first timer initialization, then the async call on which timeout is needed then async_await .